File: app/src/main/java/com/chinachino/mvvm/Requests/ServiceGenerator.java
package com.chinachino.mvvm.Requests;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static com.chinachino.mvvm.Utils.Constants.BASE_URL;

public class ServiceGenerator {
    private static Retrofit.Builder retrofitbuilder =
            new Retrofit.Builder().
                    addConverterFactory(GsonConverterFactory.create())
                    .baseUrl(BASE_URL);

    private static Retrofit retrofit = retrofitbuilder.build();
    private static TheMovieDataBaseAPI theMovieDataBaseAPI = retrofit.create(TheMovieDataBaseAPI.class);

    public static TheMovieDataBaseAPI GetMovies(){
        return theMovieDataBaseAPI;
    }
}

File: app/src/main/java/com/chinachino/mvvm/requests/ServiceGenerator.kt
package com.chinachino.mvvm.requests

import com.chinachino.mvvm.utils.Constants.BASE_URL
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ServiceGenerator {
    private val retrofitBuilder: Retrofit.Builder = Retrofit.Builder().addConverterFactory(GsonConverterFactory.create())
            .baseUrl(BASE_URL)
    private val retrofit = retrofitBuilder.build()
    private val theMovieDataBaseAPI = retrofit.create(TheMovieDataBaseAPI::class.java)
    fun getMovies(): TheMovieDataBaseAPI {
        return theMovieDataBaseAPI
    }
}